src/raft_server.cxx

```diff
@@ -837,6 +861,7 @@ void raft_server::handle_peer_resp(ptr<resp_msg>& resp, ptr<rpc_exception>& err)
             peer* pp = entry->second.get();
             int rpc_errs = pp->get_rpc_errs();
             if (rpc_errs >= raft_server::raft_limits_.warning_limit_) {
+                pp->set_recovered();
```

Comment:
`set_lost()` is invoked in two places:
1) `handle_peer_resp`: based on `raft_limits_.warning_limit_`, and
2) `check_leadership_validity`: from `get_not_responding_peers` which is based on `raft_limits_.response_limit_`.

Although both limits have the same default value, they can be configured with different numbers. If `warning_limit_ > response_limit_`, there can be a possibility that `set_recovered()` may not be called, when a leader loses its role and then becomes leader later again.

Please confirm this. I think calling `set_recovered` outside of this if-clause is ok. It is just setting an atomic variable.

Reply:
On second thought, calling `set_recovered` for all peers when a server becomes leader (in `raft_server::become_leader`) is also an option. Leader may need a fresh start.